src/main/java/uk/gov/hmcts/reform/wataskmanagementapi/services/utils/TaskMandatoryFieldsValidator.java

```diff
@@ -0,0 +1,150 @@
+                return;
+            }
+
+            JsonNode excludedJurisdictionsArray =
+                jsonParserUtils.parseJson(mandatoryFieldCheckEnabledServices.toJsonString(),
+                                          MANDATORY_FIELD_CHECK_FLAG_VARIANT.stringValue());
+            if (excludedJurisdictionsArray == null) {
+                log.warn("Excluded jurisdictions array is null, skipping jurisdiction check.");
+                validateTaskMandatoryFields(task);
+                return;
+            }
+
+            if (!isJurisdictionExcluded(task, excludedJurisdictionsArray)) {
+                validateTaskMandatoryFields(task);
+            }
+        }
+    }
+
+    /**
+     * Checks if the task's jurisdiction is excluded.
+     *
+     * @param task the task to be checked
+     * @param excludedJurisdictionsArray the array of excluded jurisdictions
+     * @return true if the jurisdiction is excluded, false otherwise
+     */
+    private boolean isJurisdictionExcluded(TaskResource task, JsonNode excludedJurisdictionsArray) {
+        for (JsonNode jurisdiction : excludedJurisdictionsArray) {
+            if (jurisdiction.asText().equals(task.getJurisdiction())) {
+                log.warn("Task {} is excluded from mandatory field check due to jurisdiction being excluded {}",
+                         task.getTaskId(), task.getJurisdiction());
+                return true;
+            }
+        }
+        return false;
+    }
+
+    /**
+     * Validates the mandatory fields of a task.
+     *
+     * @param task the task to be validated
+     * @throws ValidationException if any mandatory field is missing or invalid
+     * @throws IllegalArgumentException if a property value cannot be found
+     */
+    public void validateTaskMandatoryFields(TaskResource task) {
+        List<String> serviceSpecificErrors = new ArrayList<>();
+        List<String> tmSpecificErrors = new ArrayList<>();
+
+        for (String field : taskMandatoryFields) {
+            try {
+                Object fieldValue = PropertyUtils.getProperty(task, field);
+                if (isFieldNullOrEmpty(fieldValue)) {
+                    addError(field, serviceSpecificErrors, tmSpecificErrors);
+                }
+            } catch (Exception e) {
+                throw new IllegalArgumentException("Cannot find property value for mandatory field " + field, e);
+            }
+        }
+
+        if (!serviceSpecificErrors.isEmpty()) {
+            throw new ServiceMandatoryFieldValidationException(
+                String.join(", ", serviceSpecificErrors.toString()));
+        }
+        if (!tmSpecificErrors.isEmpty()) {
+            throw new ValidationException(String.join(", ", tmSpecificErrors.toString()));
+        }
+    }
+
+    private boolean isFieldNullOrEmpty(Object fieldValue) {
+        return fieldValue == null || fieldValue.toString().isBlank();
```

Comment:
Will the Object.toString() ever be blank?  If it's not set won't it just be an object reference representation?  Maybe you have tested that scenario?

Reply:
If the value is empty string then fieldValue.toString().isBlank() is returning true.